asf_public_discourse_home_decarbonisation/analysis/mse/initial_text_analysis_category_data.py

```diff
@@ -0,0 +1,442 @@
+"""
+This script performs an initial text analysis on Money Saving Expert data, by:
+- Computing the frenquency os specific words and n-grams of interest;
+- Identiyfying top words and n-grams;
+
+Plots are saved in the folder `asf_public_discourse_home_decarbonisation/outputs/figures/mse/`.
+
+To run this script:
+`python asf_public_discourse_home_decarbonisation/analysis/mse/initial_text_analysis_category_data.py`
+
+To change category or date/time range, use the following arguments:
+`python asf_public_discourse_home_decarbonisation/analysis/mse/initial_text_analysis_category_data.py --category <category> --collection_date_time <collection_date_time>`
+where
+<category>: category or sub-forum to be analysed, e.g. "energy", "lpg-heating-oil-solid-other-fuels", etc.
+<collection_date_time>: data collection date/time in the format YYYY_MM_DD
+"""
+
+# ## Package imports
+import warnings
+
+# Ignoring warnings from Pandas
+warnings.simplefilter(action="ignore")
+
+import numpy as np
+import pandas as pd
+from nltk.tokenize import word_tokenize
+import os
+from asf_public_discourse_home_decarbonisation.getters.mse_getters import get_mse_data
+from asf_public_discourse_home_decarbonisation.utils.text_processing_utils import (
+    process_text,
+    create_ngram_from_ordered_tokens,
+    english_stopwords_definition,
+    lemmatize_sentence,
+    frequency_ngrams,
+)
+from asf_public_discourse_home_decarbonisation.utils.plotting_utils import (
+    plot_and_save_top_ngrams,
+    plot_and_save_wordcloud,
+)
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+
+import argparse
+import logging
+
+logger = logging.getLogger(__name__)
+
+# Analysis-specific parameters
+number_ngrams_wordcloud = 30
+min_frequency_tokens = 100
+min_frequency_bigrams = 50
+min_frequency_trigrams = 10
+top_ngrams_barplot = 10
+
+# Specific groups of keywords we want to look at
+keyword_dictionary = {
+    "heat_pump_keywords": ["heat pump", "ashp", "gshp", "wshp", "air 2 air"],
+    "boiler_keywords": ["boiler"],
+    "hydrogen_keywords": ["hydrogen"],
+    "bus_keywords": ["boiler upgrade scheme", "bus"],
+    "grants_keywords": [
+        "boiler upgrade scheme",
+        "bus",
+        "renewable heat incentive",
+        "domestic rhi",
+        "clean heat grant",
+        "home energy scotland grant",
+        "home energy scotland loan",
+        "home energy scotland scheme",
+    ],
+    "mcs_keywords": ["mcs", "microgeneration certification scheme"],
+    "cost_est_keywords": ["cost estimator"],
+    "nesta_keywords": ["nesta"],
+    "installer_keywords": ["installer", "engineer"],
+    "installation_keywords": ["installation"],
+    "cost_keywords": ["cost", "price", "pay"],
```

Comment:
pound(s), the symbol £ (maybe we cleaned for this) as well?

Reply:
good shout, will add!